cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/common/MetadataClientTest.java

```diff
@@ -0,0 +1,291 @@
+    EasyMock.expect(mockClusterIdFuture.get(expectedTimeoutMs, TimeUnit.MILLISECONDS))
+        .andReturn("cluster-1");
+    EasyMock.expect(mockDescribeClusterResult.clusterId()).andReturn(mockClusterIdFuture);
+    KafkaFuture mockControllerFuture = EasyMock.mock(KafkaFuture.class);
+    EasyMock.expect(mockControllerFuture.get(expectedTimeoutMs, TimeUnit.MILLISECONDS))
+        .andReturn(new Node(0, "host0", 100));
+    EasyMock.expect(mockDescribeClusterResult.controller()).andReturn(mockControllerFuture);
+
+    KafkaFuture mockNodesFuture = EasyMock.mock(KafkaFuture.class);
+    EasyMock.expect(mockNodesFuture.get(expectedTimeoutMs, TimeUnit.MILLISECONDS))
+        .andReturn(Arrays.asList(_controllerNode, _simpleNode));
+    EasyMock.expect(mockDescribeClusterResult.nodes()).andReturn(mockNodesFuture);
+
+    EasyMock.expect(mockAdminClient.describeCluster()).andReturn(mockDescribeClusterResult);
+    EasyMock.expect(mockAdminClient.describeCluster(EasyMock.anyObject())).andReturn(mockDescribeClusterResult);
+    EasyMock.replay(mockDescribeClusterResult, mockClusterIdFuture,
+        mockControllerFuture, mockNodesFuture);
+  }
+
+  private void mockListTopics(AdminClient mockAdminClient, Collection<String> topicNames,
+                              int expectedTimeoutMs) throws InterruptedException, ExecutionException,
+      java.util.concurrent.TimeoutException {
+    ListTopicsResult mockListTopicsResult = EasyMock.mock(ListTopicsResult.class);
+    KafkaFuture mockKafkaFuture = EasyMock.mock(KafkaFuture.class);
+    EasyMock.expect(mockKafkaFuture.get(expectedTimeoutMs, TimeUnit.MILLISECONDS))
+        .andReturn(new HashSet<>(topicNames));
+    EasyMock.expect(mockListTopicsResult.names()).andReturn(mockKafkaFuture);
+
+    EasyMock.expect(mockAdminClient.listTopics(EasyMock.anyObject())).andReturn(mockListTopicsResult);
+    EasyMock.expect(mockAdminClient.listTopics()).andReturn(mockListTopicsResult);
+    EasyMock.replay(mockListTopicsResult, mockKafkaFuture);
+  }
+
+  private MockTime mockTime() {
+    return new MockTime(0, 0L,
+        TimeUnit.NANOSECONDS.convert(0L, TimeUnit.MILLISECONDS));
+  }
+
+  private static class PartitionInfoComparisonCriteria extends ComparisonCriteria {
+    @Override
+    protected void assertElementsEqual(Object expected, Object actual) {
+      PartitionInfo expectedInfo = (PartitionInfo) expected;
+      PartitionInfo actualInfo = (PartitionInfo) actual;
+      Assert.assertEquals(expectedInfo.topic(), actualInfo.topic());
+      Assert.assertEquals(expectedInfo.partition(), actualInfo.partition());
+      Assert.assertEquals(expectedInfo.leader(), actualInfo.leader());
+      Assert.assertArrayEquals(expectedInfo.replicas(), actualInfo.replicas());
+      Assert.assertArrayEquals(expectedInfo.inSyncReplicas(), actualInfo.inSyncReplicas());
+      Assert.assertArrayEquals(expectedInfo.offlineReplicas(), actualInfo.offlineReplicas());
```

Comment:
Can we add test cases to verify offline replicas, logdirs, brokers, and partitions in response?

Reply:
I don't see where we have logdirs in the Cluster object. I improved the test to check for all others